plugins/build-style-dictionary-plugin.spec.mts

```diff
@@ -152,6 +161,74 @@ describe('buildStyleDictionaryPlugin', () => {
   --sky-color-text-default: var(--rainbow-color-gray-1);
 }
 }
+`,
+    });
+  });
+
+  it('should include the correct media queries for and selectors for container breakpoints', async () => {
+    vi.spyOn(exports, 'tokenConfig', 'get').mockReturnValue({
+      rootPath: 'plugins/fixtures/',
+      tokenSets: [
+        {
+          name: 'responsive-rainbow',
+          selector: '.sky-theme-rainbow',
+          outputPath: 'responsive-rainbow.css',
+          path: 'base-rainbow.json',
+          referenceTokens: [
+            {
+              name: 'rainbow-colors-m',
+              responsive: {
+                breakpoint: 'm',
+              },
+              path: 'responsive-rainbow-colors-m.json',
+            },
+            {
+              name: 'rainbow-colors-xs',
+              responsive: {
+                breakpoint: 'xs',
+              },
+              path: 'responsive-rainbow-colors-xs.json',
+            },
+          ],
+        },
+      ],
+    });
+
+    const plugin = buildStyleDictionaryPlugin();
+    const emitFileSpy = vi.fn();
+    if (plugin.generateBundle) {
+      await plugin.generateBundle.call({
+        emitFile: emitFileSpy,
+      });
+    }
+
+    expect(emitFileSpy).toHaveBeenCalledOnce();
+    expect(emitFileSpy).toHaveBeenCalledWith({
+      type: 'asset',
+      fileName: 'assets/scss/responsive-rainbow.css',
+      source: `.sky-theme-rainbow {
+  --rainbow-color-gray-1: #e2e3e7;
+  --rainbow-color-gray-2: #c0c2c5;
+  --rainbow-color-red-1: #fc0330;
+  --rainbow-color-red-2: #8a2538;
+  --rainbow-space-s: 10px;
+}
+.sky-theme-rainbow {
+  --sky-color-text-default: var(--rainbow-color-red-1);
+}
+.sky-theme-rainbow .sky-responsive-container-xs, .sky-theme-rainbow .sky-responsive-container-sm, .sky-theme-rainbow .sky-responsive-container-md, .sky-theme-rainbow .sky-responsive-container-lg {
```

Comment:
For xs only - is it possible to combine the first block with the responsive container block? 

I know it isn't possible for the others due to the media query - but for xs it will be the same.